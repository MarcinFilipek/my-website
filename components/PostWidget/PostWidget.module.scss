@import "../../theme/theme.scss";

.container {
  width: 100%;
  background: $fill;
  border: $border;
  box-shadow: $shadow;
  border-radius: 12px;
  display: flex;
  flex-direction: column;
  padding: 16px;
}

.title {
  color: $header;
  font-weight: 600;
  font-size: 1.2rem;
}

.description {
  color: $body1;
  font-weight: 200;
  font-size: 0.8rem;
}
