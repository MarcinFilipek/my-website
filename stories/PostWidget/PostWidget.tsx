import React from "react";
import { Typography, useTheme } from "@mui/material";

type Props = {
  title: string;
};

export const PostWidget = ({ title }: Props) => {
  const theme = useTheme();
  return (
    <div
      style={{
        display: "flex",
        flexDirection: "column",
        width: "100%",
        padding: "1rem",
        border: `1px solid ${theme.palette.primary.main}66`,
        background: "#100d21",
        borderRadius: "0.8rem",
        boxShadow: `0px 4px 24px ${theme.palette.primary.main}26`,
      }}
    >
      <Typography variant="h6" style={{ color: theme.palette.primary.main }}>
        {title}
      </Typography>
      <Typography
        variant="body2"
        style={{ color: theme.palette.secondary.main }}
      >
        Lorem Ipsum is simply dummy text of the printing and typesetting
        industry. Lorem Ipsum has been the industry's standard dummy text ever
        since the 1500s, when an unknown printer took a galley of type and
        scrambled it to make a type specimen book.
      </Typography>
    </div>
  );
};
