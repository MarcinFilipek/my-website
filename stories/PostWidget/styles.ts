import { styled, Typography } from "@mui/material";

export const Container = styled("div")(
  ({ theme }) => `
        display: flex;
        flex-direction: column;
        width: 100%;
        padding: 1rem;
        border: 1px solid ${theme.palette.primary.main}66;
        background: #100d21;
        border-radius: 0.8rem;
        box-shadow: 0px 4px 24px ${theme.palette.primary.main}26;
        `
);

export const Title = styled(Typography)(
  ({ theme }) => `color: ${theme.palette.primary.main};`
);

export const Description = styled(Typography)(
  ({ theme }) => `color: ${theme.palette.secondary.main};`
);
