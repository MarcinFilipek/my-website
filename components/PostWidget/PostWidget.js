import styles from "./PostWidget.module.scss";

export const PostWidget = () => (
  <div className={styles.container}>
    <span className={styles.title}>Title</span>
    <p className={styles.description}>Note</p>
  </div>
);
