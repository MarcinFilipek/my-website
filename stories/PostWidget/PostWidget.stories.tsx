import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";

import { PostWidget } from "./PostWidget";

const Template: ComponentStory<typeof PostWidget> = (args) => (
  <div
    style={{
      width: "500px",
      height: "100%",
      padding: "20px",
    }}
  >
    <PostWidget {...args} />
  </div>
);

export const Default = Template.bind({});

Default.args = { title: "Title" };

export default {
  title: "PostWidget",
  component: PostWidget,
} as ComponentMeta<typeof PostWidget>;
