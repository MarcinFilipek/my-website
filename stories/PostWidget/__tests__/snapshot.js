import { render } from "@testing-library/react";
import { PostWidget } from "../PostWidget";

it("renders homepage unchanged", () => {
  const { container } = render(<PostWidget title={"Title"} />);
  expect(container).toMatchSnapshot();
});
