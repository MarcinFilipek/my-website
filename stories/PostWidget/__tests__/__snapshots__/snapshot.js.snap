// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    style="display: flex; flex-direction: column; width: 100%; padding: 1rem; border: 1px solid #1976d266; background: rgb(16, 13, 33); border-radius: 0.8rem; box-shadow: 0px 4px 24px #1976d226;"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      data-testid="title"
      style="color: rgb(25, 118, 210);"
    >
      Title
    </h6>
    <p
      class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      style="color: rgb(156, 39, 176);"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
    </p>
  </div>
</div>
`;
